src/main/java/com/minecolonies/blockout/controls/ItemIcon.java

```diff
@@ -72,19 +73,22 @@ protected void drawSelf(final int mx, final int my)
      */
     private void drawItemStack(final ItemStack stack, final int x, final int y)
     {
+        if (ItemStackUtils.isEmpty(itemStack))
+        {
+            return;
+        }
+
         final RenderItem itemRender = mc.getRenderItem();
 
-        GlStateManager.translate(0.0F, 0.0F, GUI_ITEM_Z_TRANSLATE);
-        this.zLevel = GUI_ITEM_Z_LEVEL;
-        itemRender.zLevel = GUI_ITEM_Z_LEVEL;
+        GlStateManager.translate(x, y, GUI_ITEM_Z_TRANSLATE);
+        GlStateManager.scale(this.getWidth() / 16f, this.getHeight() / 16f, 1f);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16f to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16f to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS818)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS818)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS818)